### PR TITLE
IA-2891 IA-2880 fix OOM issue for getRuntime

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-2891
https://broadworkbench.atlassian.net/browse/IA-2880 

* We don't expose dataproc instances info in `getRuntime`, which can cause OOM for leo when the dataproc cluster is big. Hence refactor the DB query to not bother joining instance table
* Fix the SSD bug...Tested in fiab

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by xuscrbw — APPROVED** [review]
LGTM.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeServiceDbQueries.scala [hunk 1]
```diff
@@ -44,28 +46,114 @@ object RuntimeServiceDbQueries {
...
+       Option[ScopeRecord],
+       Option[PatchRecord])
+    ],
+    runtimeConfig: RuntimeConfig,
+    diskConfig: Option[DiskConfig]
+  ): Seq[GetRuntimeResponse] = {
+    // Call foldMap to aggregate a flat sequence of (cluster, instance, label) triples returned by the query
+    // to a grouped (cluster -> (instances, labels)) structure.
+    // Note we use Chain instead of List inside the foldMap because the Chain monoid is much more efficient than the List monoid.
+    // See: https://typelevel.org/cats/datatypes/chain.html
```
Comment (kyuksel) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeServiceDbQueries.scala:87]: good to know!

**Comment by Qi77Qi** [comment]
feel free to leave comments and I'll address them in a follow-up PR if needed..but think this is good to merge

**Review by kyuksel — APPROVED** [review]
LGTM 👍🏼  
Thanks for fixing these!

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeServiceDbQueries.scala (hunk 1 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeServiceDbQueries.scala:87]: I think Rob probably added those initially..I just copied it over :)

**Review by rtitle — COMMENTED** [review]
👍  - thanks for fixing!

**Pull request merged**